### PR TITLE
Spess Knife as a random mail item for engi and clown

### PR DESCRIPTION
**Pull request by MTandi**
## About The Pull Request

Added Spess Knife (introduced in #75537) to the random mail for station engineers and clowns.

## Why It's Good For The Game

It was added as a rare maintenance loot and barely been used.

## Changelog

:cl:
add: There is a chance that Spess Knife will be in a mail for engi and clowns
/:cl:

### CODE REVIEW

**Comment by Jacquerel** [comment]
What is a space knife?
Why is it useful for miners?
Why is it bad for something to be uncommonly used?

"This rare item only appears rarely" is not adequate justification for making a change. We _like_ unique items that don't reliably show up every round.

**Comment by MTandi** [comment]
> What is a space knife?

A multi-tool of Knife, Screwdriver and Wirecutters combined with a 10% (50% for clumsy) chance to extend some other tools:

```
	tool_behaviour = pick_weight(list(
		TOOL_KNIFE = 10,
		TOOL_SCREWDRIVER = 10,
		TOOL_WIRECUTTER = 10,
		TOOL_WRENCH = 5,
		TOOL_SHOVEL = 2,
		TOOL_SAW = 2,
		TOOL_ROLLINGPIN = 1,
	))
```

> Why is it useful for miners? 

It's a swiss army knife prototype that is a survival tool, perfect for miners. Can argue about bitrunners.

> Why is it bad for something to be uncommonly used?

I've never seen this in-game and people don't even know it exists, rare maint loot is too rare to obtain it. Needs some semi-reliable way of obtaining.

> "This rare item only appears rarely" is not adequate justification for making a change. We _like_ unique items that don't reliably show up every round.

Please suggest something other than random maint spawn then.

**Comment by Jacquerel** [comment]
What does a miner need tools for? 
If you want this to just be generically available then _I guess_ add it as a cargo goodie, or in the mail loot pool for certain jobs.

**Comment by carlarctg** [comment]
I get what youre saying jacq but the spess knife is comically, absurdly rare. I've pretty much never seen it in a round, and it's really not interesting enough an item to warrant its absurdly low chances of existing in a shift.

**Comment by Jacquerel** [comment]
> I get what youre saying jacq but the spess knife is comically, absurdly rare. I've pretty much never seen it in a round, and it's really not interesting enough an item to warrant its absurdly low chances of existing in a shift.

I don't have an inherent objection to it appearing more often (especially as the PR is from the person who added it) but don't think the mining vendor is the correct venue, the PR as it is now is better to me.

**Comment by optimumtact** [comment]
>it's really not interesting enough an item to warrant its absurdly low chances of existing in a shift.

It's effectively a toolbelt which is still pretty valuable.

**Comment by Livrah** [comment]
If anyone, the assistant

**Comment by kawaiinick** [comment]
why clown?

**Comment by MTandi** [comment]
> If anyone, the assistant

Don't want to make it an incentive to pick assistant.

> why clown?

Because the clumsy trait makes the knife more likely to give a tool, different from what you select.

**Comment by HoodMilk** [comment]
I have seen a Spess Knife literally once over like 500 hours of playing with it ingame, absurdly rare item for no good reason, it's just a neat utility. If anything you should be able to buy them from cargo as a goodie.

**Comment by Loafin34** [comment]
let us buy it from the black market as well please

**Comment by MTandi** [comment]
> let us buy it from the black market as well please

I'd rather make it a spy bounty so that it appears on black market when stolen. So that it is a bit more unique. But that will need adding it somewhere roundstart, so unlikely.

File: code/modules/jobs/job_types/clown.dm [hunk 1]
```diff
@@ -104,3 +104,8 @@
 	var/datum/atom_hud/fan = GLOB.huds[DATA_HUD_FAN]
 	fan.show_to(H)
 	H.faction |= FACTION_CLOWN
+
+/datum/job/clown/get_mail_goodies(mob/recipient)
+	. = ..()
+	// As a joke for clumsy clown from engineering department
+	. += list(/obj/item/spess_knife = 10)
```
Comment (MrMelbert) [inline code/modules/jobs/job_types/clown.dm:111]: Why is this being done in the proc and not just in the `mail_goodies` defined list

File: code/modules/jobs/job_types/station_engineer.dm [hunk 2]
```diff
@@ -78,3 +78,8 @@
 	head = null
 	mask = /obj/item/clothing/mask/breath
 	internals_slot = ITEM_SLOT_SUITSTORE
+
+/datum/job/engineer/get_mail_goodies(mob/recipient)
+	. = ..()
+	// A multi-tool for station work
+	. += list(/obj/item/spess_knife = 20)
```
Comment (MrMelbert) [inline code/modules/jobs/job_types/station_engineer.dm:85]: Same

**Review by MrMelbert — CHANGES_REQUESTED** [review]

**Pull request merged**